src/script/lua_api/l_util.cpp

```diff
@@ -329,6 +329,51 @@ int ModApiUtil::l_mkdir(lua_State *L)
 	return 1;
 }
 
+// rmdir(path, recursive)
+int ModApiUtil::l_rmdir(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	const char *path = luaL_checkstring(L, 1);
+	CHECK_SECURE_PATH(L, path, true);
+
+	bool recursive = false;
+
+	if (!lua_isnil(L, 2)) {
+		recursive = lua_toboolean(L, 2);
+	}
+
+	if (recursive) {
+		lua_pushboolean(L, fs::RecursiveDelete(path));
+	} else {
+		lua_pushboolean(L, fs::DeleteSingleFileOrEmptyDirectory(path));
+	}
+	return 1;
+}
+
+// cpdir(source, destination, remove_source)
+int ModApiUtil::l_cpdir(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	const char *source = luaL_checkstring(L, 1);
+	const char *destination = luaL_checkstring(L, 2);
+	CHECK_SECURE_PATH(L, destination, true);
```

Comment:
check source

Reply:
Sorry, what exactly do you mean?

Reply:
To check whether `source` is a Minetest-safe path. So not every file can be copied into the Minetest structure

Reply:
Ah, I understand. I had considered that, but, do we really need to ensure that the source dir is within MT? I mean, can't you use io.open to read even the stuff outside the Minetest directory? If you can't, then OFC it would make sense to check the source, but otherwise, it really wouldn't. I don't really see any major dangers without copying from outside of MT's structure to begin with though.

Reply:
Just tested, and I guess you can't read from outside of the MT dir. It is worth noting that you can copy directories from non-Minetest locations in the main menu (66c4108581902a34e2e42a4551c8c1157a881681), but I only implemented this so that mods could be installed from ZIP files. And, it would probably be more dangerous for normal mods, considering that they're not like the main menu that just ships with the game.

With this in mind, I'll add a check to ensure that the source is within the MT dir.

Reply:
The main menu isn't sandboxed